src/OpenRiaServices.Hosting.Wcf/Framework/WCF/Behaviors/DomainOperationInvoker.cs

```diff
@@ -112,11 +115,15 @@ private async ValueTask<object> InvokeAsync(object instance, object[] inputs)
             }
         }
 
-        private DomainService GetDomainService(object instance, WcfDomainServiceContext context)
+        private DomainService GetDomainService(DomainServiceBehavior.DomainServiceInstanceInfo instanceInfo, WcfDomainServiceContext context)
         {
             // create and initialize the DomainService for this request
-            DomainServiceBehavior.DomainServiceInstanceInfo instanceInfo =
-                (DomainServiceBehavior.DomainServiceInstanceInfo)instance;
+            if (instanceInfo.ServiceScope.ServiceProvider.GetService(instanceInfo.DomainServiceType) is DomainService service)
+            {
+                // Do NOT save instance in instanceInfo.DomainServiceInstance since container will dispose instance
+                service.Initialize(context);
```

Comment:
If i remember correcly Initialize throws exception if the service is already initialized, could this happen here? Probably won't try to get the same DomainService twice

Reply:
Good catch, that might happen if someone registers a service as a Singleton (or implements pooling)

It sounds like it could be somewhat unsafe since ServiceContext.User is used for authentication. 
I assume that if anyone does this via the DomainServiceFactory they will override Intitialize and make it work for their scenario.